tests/functional/docs/test_iam.py

```diff
@@ -0,0 +1,68 @@
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from tests.functional.docs import BaseDocsFunctionalTest
+from botocore.docs.service import ServiceDocumenter
+
+
+class TestIAMDocs(BaseDocsFunctionalTest):
+    def test_changes_policy_doc_to_dict(self):
+        modified_methods = ['create_instance_profile',
+                            'create_policy',
+                            'create_policy_version',
+                            'create_role',
+                            'get_account_authorization_details',
+                            'get_group_policy',
+                            'get_instance_profile',
+                            'get_policy_version',
+                            'get_role',
+                            'get_role_policy',
+                            'get_user_policy',
+                            'list_instance_profiles',
+                            'list_instance_profiles_for_role',
+                            'list_policy_versions',
+                            'list_roles',
+                            'put_group_policy',
+                            'put_role_policy',
+                            'put_user_policy',
+                            'update_assume_role_policy']
+        service_contents = ServiceDocumenter('iam', self._session)\
+            .document_service()
+        for method_name in modified_methods:
+            method_contents = self.get_method_document_block(
+                method_name, service_contents)
+            self.assertNotIn(b"Document': 'string'", method_contents)
+            self.assertNotIn(b"Document='string", method_contents)
+            self.assertTrue(
+                (b"Document': {}" in method_contents) or
+                (b"Document={}" in method_contents))
+            self.assertNotIn(b"Document** *(string)", method_contents)
+            self.assertNotIn(b"Document: string", method_contents)
+            self.assertTrue(
+                (b"Document** *(dict)" in method_contents) or
+                (b"Document: dict" in method_contents))
+
+    def test_changes_policy_doc_list_to_dict(self):
+        modified_methods = ['get_context_keys_for_custom_policy',
+                            'get_context_keys_for_principal_policy',
+                            'simulate_custom_policy',
+                            'simulate_principal_policy']
+        service_contents = ServiceDocumenter('iam', self._session)\
+            .document_service()
+        for method_name in modified_methods:
+            method_contents = self.get_method_document_block(
+                method_name, service_contents)
+            self.assertNotIn(
```

Comment:
I would be hesitant to have newlines in the output. It may cause failures on windows because carriage returns are sometimes added to the end. I am not completely sure about that though. I would prefer using ``get_parameter_document_block`` and then ``assert_contains_lines_in_order`` and ``assert_not_contains_line``